pkg/allocateip/vxlan.go

```diff
@@ -0,0 +1,34 @@
+package allocateip
+
+import (
+	api "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/net"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// determineVXLANEnabledPools returns all VXLAN enabled pools.
+func determineVXLANEnabledPoolCIDRs(node api.Node, ipPoolList api.IPPoolList) []net.IPNet {
+	var cidrs []net.IPNet
+	for _, ipPool := range ipPoolList.Items {
+		_, poolCidr, err := net.ParseCIDR(ipPool.Spec.CIDR)
+		if err != nil {
+			log.WithError(err).Fatalf("Failed to parse CIDR '%s' for IPPool '%s'", ipPool.Spec.CIDR, ipPool.Name)
+		}
+
+		// Check if IP pool selects the node
+		if selects, err := ipPool.SelectsNode(node); err != nil {
```

Comment:
Worth making a utility function `getIPPoolsThatApplyToThisNode` to share this filtering logic between IPIP and VXLAN?

Reply:
yeah, might be able to commonize this whole function